api/metadata/models.py

```diff
@@ -0,0 +1,44 @@
+from core.models import AbstractBaseExportableModel
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from organisations.models import Organisation
+
+FIELD_DATA_MAX_LENGTH = 2000
+
+
+class FieldType(models.TextChoices):
+    INTEGER = "int"
+    STRING = "str"
+    BOOLEAN = "bool"
+    FLOAT = "float"
+
+
+class MetadataField(AbstractBaseExportableModel):
+    name = models.CharField(max_length=255)
+    type = models.CharField(
+        max_length=255, choices=FieldType.choices, default=FieldType.STRING
+    )
+    description = models.TextField(blank=True, null=True)
+    organisation = models.ForeignKey(Organisation, on_delete=models.CASCADE)
+
+
+class MetadataModelField(AbstractBaseExportableModel):
+    field = models.ForeignKey(MetadataField, on_delete=models.CASCADE)
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    is_required = models.BooleanField(default=False)
+
+    class Meta:
+        unique_together = ("field", "content_type")
+
+
+class Metadata(AbstractBaseExportableModel):
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
```

Comment:
... what I don't understand is why we need content_type here when it's also defined against the MetadataModelField class (which we also link to below)?

Reply:
Yeah, it's for creating the `GenericForeignKey` which is used for creating a reverse relation here:
https://github.com/Flagsmith/flagsmith/pull/1812/files#diff-a0b057e41b454fcf5879b6c8ddbb1cb10a4c446e671ef37ff3b242efa1e7200bR84

Reply:
Is there any way around it? Can the reverse relation be created based on related fields for example? Something like `model_field__content_type`. Feels weird that we're having to define the same data twice.